src/Core/Content/Mail/Message/SendMailHandler.php

```diff
@@ -0,0 +1,63 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Content\Mail\Message;
+
+use League\Flysystem\FilesystemOperator;
+use Psr\Log\LoggerInterface;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Component\Mailer\Exception\TransportExceptionInterface;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+use Symfony\Component\Messenger\Attribute\AsMessageHandler;
+use Symfony\Component\Mime\Email;
+
+/**
+ * @internal
+ */
+#[AsMessageHandler]
+#[Package('services-settings')]
+final class SendMailHandler
+{
+    /**
+     * @internal
+     */
+    public function __construct(
+        private readonly TransportInterface $transport,
+        private readonly FilesystemOperator $filesystem,
+        private readonly LoggerInterface $logger
+    ) {
+    }
+
+    /**
+     * @throws TransportExceptionInterface
+     */
+    public function __invoke(SendMailMessage $message): void
+    {
+        $mailDataPath = $message->getMailDataPath();
+
+        if (!$this->filesystem->fileExists($mailDataPath)) {
+            $this->logger->error('An error occurred while sending an email. The mail data file does not exist.');
```

Comment:
we should include the path here for context to make debugging easier, and probably also document why we don't throw here as it does not make sense to retry that message right? 

Or should we rather use an exception and mark it as non-retryable? https://symfony.com/doc/current/messenger.html#avoiding-retrying